tests/Fixer/Operator/BinaryOperatorSpacesFixerTest.php

```diff
@@ -474,7 +474,7 @@ public function provideConfiguredCases(): array
             ],
             'align array destruction' => [
                 '<?php
-                    $c = [$d] = $e[1];
+                    $c                 = [$d]                 = $e[1];
```

Comment:
the placement of `$e[1]` is this really better than before?

Reply:
Fair enough, I think we shouldn't merge utest that prove an issue though. There are a bunch of issues reported on this rule and PR's with tests to prove these issues. I think going slow on merging changes on this rule might be good, because of the number of open issues, that is all.

Reply:
hmm testing it again, your PR fixes your issue but the fix does introduce this side effect of:
```diff
-                    $c                 = [$d] = $e[1];
+                    $c = [$d] = $e[1];
                     function A(){}[$a] = $a[$c];
                     $b                 = 1;

```
We can talk about if it is good or bad, but the code that was added was designed to not align the `=` here.

Reply:
This was introduced by https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2475
and the test is called testAlignArrayDestruction.

I clearly don't understand what the test was looking for. `align` is supposed to align all the following `=` of the same scope. And we're in the same scope here. So yes, they need to be aligned and the behavior is not supposed to be discussed here.

The only discussion is "what about the next = ?". 
And I think it should be
```
$c                 = [$d] = $e[1];
function A(){}[$a] = $a[$c];
$b                 = 1;
```
and we should ignore the same `=` of the same line.

Reply:
I proposed a fix for this situation https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6450

I'm currently using the `align` on several projects and my current understanding is the following:
"If two identical binary operator are on the following lines and in the same scope, the need to be aligned".

I cannot believe that this behavior is correct
```
$c = [$d] = $e[1];
function A(){}[$a] = $a[$c];
$bbbb              = 1;
```
it should be treated the same way than
```
$c     = [$d] = $e[1];
$d[$a] = $a[$c];
$bbbb  = 1;
```

The main issue was the fact we ended with 
```
$c                 = [$d]                 = $e[1];
```
and this is fixed by https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6450